### PR TITLE
Code Review : React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by amiraabouhadid**
## Description

This is a web app where users can view top, new, and staff picks albums from Napster's API. Users can also view an album's tracks and even download and listen to a track preview! 

You can watch a walkthrough of the application here [video](https://www.loom.com/share/9e0387f5180b474eb17d98fbe4808851)

### General requirements
- [x] There are no linter errors.
- [x] Followed the **Github flow**.
- [x] The documentation is professional.
### HTML/CSS/JS requirements

- [x] We followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- [x] We followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).
### Project requirements
Interactions
- Home page
    - When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
    - Along with the category name, you will display some numeric values per category that come from the API.
    - When the user clicks (or taps) on a category item, the application navigates to the details page.
- Details page
    - In the details page, the webapp retrieves data from the API to show detailed data in the category.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.

Testing requirements
- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.
    - You may need to mock the access to the API, so that your tests don't send actual requests.
    - You may need to mock the connection to the Redux Store.

Technical requirements
- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- You should filter the data that you retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- The project should be deployed and accessible online (add a link to the online version of your app to your README file).

Thank you for taking the time to review this pull request.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,21 @@
...
-          Learn React
-        </a>
-      </header>
-    </div>
+
+    <Routes>
+      <Route exact path="/music_to_my_ears/" element={<Welcome />} />
+      <Route exact path="/music_to_my_ears/albums/:criteria" element={<Home />} />
+      <Route exact path="/music_to_my_ears/albums/:criteria/:id" element={<Tracks />} />
+    </Routes>
```
Comment (mateo951) [inline src/App.js:17]: The project contains a `Home` page and a `Detail` page 🤓 Well done!

File: src/App.js [hunk 2]
```diff
@@ -1,24 +1,21 @@
-import logo from './logo.svg';
-import './App.css';
+import React from 'react';
+import {
+  Routes, Route,
+} from 'react-router-dom';
```
Comment (mateo951) [inline src/App.js:4]: You successfully created a  single page application (SPA) using `react-router-dom` to serve each page without reloading 🙌🔥

File: src/App.js [hunk 3]
```diff
@@ -1,24 +1,21 @@
...
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+
+    <Routes>
+      <Route exact path="/music_to_my_ears/" element={<Welcome />} />
+      <Route exact path="/music_to_my_ears/albums/:criteria" element={<Home />} />
+      <Route exact path="/music_to_my_ears/albums/:criteria/:id" element={<Tracks />} />
```
Comment (mateo951) [inline src/App.js:16]: The category item detail page has its unique route within the SPA 🚀

File: src/index.js [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
+      <App />
+    </Router>
 
-// If you want to start measuring performance in your app, pass a function
-// to log results (for example: reportWebVitals(console.log))
-// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
-reportWebVitals();
+  </Provider>,
+  document.getElementById('root'),
+);
```
Comment (mateo951) [inline src/index.js:20]: The project uses Redux for state management and data retrieved from the API are stored in the Redux store correctly 🤓

File: src/App.test.js [hunk 5]
```diff
@@ -1,8 +1,177 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
+// import { render, screen } from '@testing-library/react';
```
Comment (mateo951) [inline src/App.test.js:1]: Your app has unit tests for several components of the app and some pure functions in Redux code! Fantastic job 🔥

**Review by mateo951 — CHANGES_REQUESTED** [review]
### Hi @amiraabouhadid, 

📝👌 **Your project highlights** 😎 
- General Requirements 💪🤓
- Great Video Presentation ✔
- App deployed using Github Pages 🚀
- UI Design 👌
- Implementation of API 🤯

Excelente work! I had a great time reviewing your project 😎 Really liked the music API you chose and the final product 🙌
There is one more issue that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

File: README.md [hunk 6]
```diff
@@ -1,70 +1,83 @@
...
 
-## Learn More
+- GitHub: [@amiraabouhadid](https://github.com/amiraabouhadid)
+- Twitter: [@AbouhadidAmira](https://twitter.com/AbouhadidAmira)
+- LinkedIn: [amira-abouhadid](https://www.linkedin.com/in/amira-abouhadid/)
 
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
 
-To learn React, check out the [React documentation](https://reactjs.org/).
+## Contributors
```
[the remark this answers is not in the document]
Comment (amiraabouhadid) [reply README.md:71]: Noted. Thanks!

File: src/App.js (hunk 1 above)
Comment (amiraabouhadid) [reply src/App.js:17]: Thank you!

File: src/App.js (hunk 2 above)
Comment (amiraabouhadid) [reply src/App.js:4]: Thank you! I am glad you like it

File: src/App.test.js (hunk 5 above)
Comment (amiraabouhadid) [reply src/App.test.js:1]: Thanks! I appreciate it

**Review by rotimiazeez — APPROVED** [review]
# Hi @amiraabouhadid , 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**